data-prepper-plugins/anomaly-detector-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/anomalydetector/modes/RandomCutForestModeTests.java

```diff
@@ -205,6 +207,7 @@ void testRandomCutForestModeVerboseFalse() {
         }
 
         final List<Record<Event>> anomalyRecords = randomCutForestMode.handleEvents(recordsWithAnomaly).stream().collect(toList());;
-        assertThat(anomalyRecords.size(), equalTo(1));
+        // Due to inherent variance in the RCF algorithm, 1-3 anomalies will be detected after the level shift.
+        assertThat(anomalyRecords.size(), both(greaterThan(0)).and(lessThan(4)));
```

Comment:
This would probably be somewhat more readable as it would have the actual valid values listed.

```
both(greaterThanOrEqualTo(1)).and(lessThanOrEqualTo(3))
```

Reply:
Good tip. Made this change, and added a small change to how metrics for this plugin are reported.